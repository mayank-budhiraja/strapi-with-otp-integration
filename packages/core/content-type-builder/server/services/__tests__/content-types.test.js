'use strict';

const { formatContentType } = require('../content-types')();

describe('Content types service', () => {
  describe('format ContentType', () => {
    const contentType = {
      uid: 'test-uid',
      kind: 'singleType',
      plugin: 'some-plugin',
      modelName: 'my-name',
      collectionName: 'tests',
      info: {
        displayName: 'My name',
        singularName: 'my-name',
        pluralName: 'my-names',
        description: 'My description',
      },
      options: {
        draftAndPublish: false,
      },
      pluginOptions: {
        'content-manager': {
          visible: true,
        },
      },
      attributes: {
        title: {
          type: 'string',
        },
      },
    };

    it('Returns consistent schemas', () => {
      expect(formatContentType(contentType)).toMatchSnapshot();
    });

    it('Sets default kind', () => {
      expect(
        formatContentType({
          ...contentType,
          kind: undefined,
        })
      ).toMatchObject({
        schema: {
          kind: 'collectionType',
        },
      });
    });
  });
});
