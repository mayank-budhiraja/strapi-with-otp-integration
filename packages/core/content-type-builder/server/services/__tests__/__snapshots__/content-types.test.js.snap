// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content types service format ContentType Returns consistent schemas 1`] = `
Object {
  "apiID": "my-name",
  "plugin": "some-plugin",
  "schema": Object {
    "attributes": Object {
      "title": Object {
        "type": "string",
      },
    },
    "collectionName": "tests",
    "description": "My description",
    "displayName": "My name",
    "draftAndPublish": false,
    "kind": "singleType",
    "pluginOptions": Object {
      "content-manager": Object {
        "visible": true,
      },
    },
    "pluralName": "my-names",
    "restrictRelationsTo": null,
    "singularName": "my-name",
    "visible": true,
  },
  "uid": "test-uid",
}
`;
